ios/Capacitor/Capacitor/Plugins/Http/Http.swift

```diff
@@ -0,0 +1,395 @@
+import Foundation
+import AudioToolbox
+
+@objc(CAPHttpPlugin)
+public class CAPHttpPlugin: CAPPlugin {
+
+  @objc public func request(_ call: CAPPluginCall) {
+    guard let urlValue = call.getString("url") else {
+      return call.reject("Must provide a URL")
+    }
+    guard let method = call.getString("method") else {
+      return call.reject("Must provide a method. One of GET, DELETE, HEAD PATCH, POST, or PUT")
+    }
+    
+    let headers = (call.getObject("headers") ?? [:]) as [String:String]
+    
+    let params = (call.getObject("params") ?? [:]) as [String:String]
+    
+    guard var url = URL(string: urlValue) else {
+      return call.reject("Invalid URL")
+    }
+    
+    
+    switch method {
+    case "GET", "HEAD":
+      get(call, &url, method, headers, params)
+    case "DELETE", "PATCH", "POST", "PUT":
+      mutate(call, url, method, headers)
+    default:
+      call.reject("Unknown method")
+    }
+  }
+
+  
+  @objc public func downloadFile(_ call: CAPPluginCall) {
+    guard let urlValue = call.getString("url") else {
+      return call.reject("Must provide a URL")
+    }
+    guard let filePath = call.getString("filePath") else {
+      return call.reject("Must provide a file path to download the file to")
+    }
+    
+    let fileDirectory = call.getString("fileDirectory") ?? "DOCUMENTS"
+    
+    guard let url = URL(string: urlValue) else {
+      return call.reject("Invalid URL")
+    }
+    
+    let task = URLSession.shared.downloadTask(with: url) { (downloadLocation, response, error) in
+      if error != nil {
+        CAPLog.print("Error on download file", downloadLocation, response, error)
+        call.reject("Error", error, [:])
```

Comment:
@mlynch I had to update all instances of `call.reject` to `call.error` when passing the error as an argument. Looks like #2533 changed that functionality.

Reply:
Seeing similar problems in `PluginCall.java`. Probably just upstream changes that broke some things in your branch.